dev-lua/lua-utf8/lua-utf8-0.1.1.ebuild

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit toolchain-funcs
+
+DESCRIPTION="UTF-8 support for Lua"
+HOMEPAGE="https://github.com/starwing/luautf8"
+SRC_URI="https://github.com/starwing/luautf8/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="luajit test"
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="virtual/pkgconfig"
+RDEPEND="
+	!luajit? ( >=dev-lang/lua-5.1:= )
+	luajit? ( dev-lang/luajit:2 )"
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/${PN//-/}-${PV}"
+
+src_prepare() {
+	default
+
+	cat > Makefile <<-EOF
+		# this file builds lua-utf8
+
+		LUA_LIBDIR      := "${ED}/$($(tc-getPKG_CONFIG) --variable INSTALL_CMOD $(usex luajit 'luajit' 'lua'))"
+		LUA_INC         := "${ED}/$($(tc-getPKG_CONFIG) --variable INSTALL_INC $(usex luajit 'luajit' 'lua'))"
+		LUA_SHARE       := "${ED}/$($(tc-getPKG_CONFIG) --variable INSTALL_LMOD $(usex luajit 'luajit' 'lua'))"
+		CWARNS          := -Wall -pedantic
+		CFLAGS          += -I\$(LUA_INC) -fPIC \$(CWARNS)
+		LIB_OPTION      := -shared
+
+		SONAME          := lua-utf8.so
+		SONAMEV         := \$(SONAME).0
+		LIBRARY         := \$(SONAMEV).1.1
+		SRC             := lutf8lib.c
+		OBJ             := \$(patsubst %.c, %.o, \$(SRC))
+
+		FILES           := parseucd.lua
+
+		all: \$(LIBRARY) \$(SONAMEV) \$(SONAME)
+
+		\$(SONAMEV):
+		TABln -s \$(LIBRARY) \$@
+
+		\$(SONAME):
+		TABln -s \$(SONAMEV) \$@
+
+		\$(LIBRARY): \$(OBJ)
+		TAB\$(CC) \$(CFLAGS) \$(LIB_OPTION) -o \$(LIBRARY) \$(OBJ) -lc
+
+		install:
+		TABmkdir -p \$(LUA_LIBDIR)
+		TABcp \$(SONAME) \$(LUA_LIBDIR)
+		TABmkdir -p \$(LUA_SHARE)
+		TABcp \$(FILES) \$(LUA_SHARE)
+
+		clean:
+		TAB\$(RM) \$(LIBRARY) \$(SONAMEV) \$(SONAME) *.o
+	EOF
```

Comment:
maybe instead of writing Makefile in heredoc, it's just better to add it to files/lua-utf8.Makefile ?
seems brittle to me.

Reply:
done